osgbuild/fetch_sources.py

```diff
@@ -19,6 +21,64 @@
 log = logging.getLogger('osgbuild')
 log.propagate = False
 
+def process_meta_url(line, destdir):
+    """
+    Process a JSON-serialized URL spec.  Should be of the format:
+     type=git url=https://github.com/opensciencegrid/cvmfs-config-osg.git name=cvmfs-config-osg tag=0.1 hash=e2b54cd1b94c9e3eaee079490c9d85f193c52249
+    'name' can be derived from the URL if the last component in the URL is of the form 'NAME.git'
+    """
+    contents = {}
+    for entry in line.split():
+        info = entry.split("=", 1)
+        if len(info) > 1:
+            contents[info[0].strip()] = info[1].strip()
+    tag_type = contents.get("type", "")
+    if tag_type != "git":
+        raise Error("Only 'git'-type URLs are understood in JSON: %s" % line)
+    git_url = contents.get('url')
+    if not git_url:
+        raise Error("No git URL provided in JSON: %s" % line)
+    name = contents.get("name")
+    if not name:
+        basename = os.path.split(git_url)[-1]
+        if basename[-4:] == '.git':
+            name = basename[:-4]
+        else:
+            raise Error("No package name specified in JSON: %s" % line)
+    print "Checking out git repo for %s." % name
+    tag = contents.get("tag")
+    if not tag:
+        raise Error("No package tag specified: %s" % line)
+    dest_file = str("%s-%s.tar.gz" % (name, tag))
```

Comment:
* Remove unnecessary use of `str()`
* Strip off leading "v" from tag name if it matches `^v\d` for generating tarball name